sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -327,6 +327,13 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val PARQUET_RECORD_FILTER_ENABLED = buildConf("spark.sql.parquet.recordLevelFilter.enabled")
+    .doc("If true, enables Parquet's native record-level filtering using the pushed down " +
+      "filters. This configuration only has an effect when 'spark.sql.parquet.filterPushdown' " +
+      "is enabled.")
+    .booleanConf
+    .createWithDefault(true)
```

Comment:
How about make the default value `false`? Since we'll always do record filtering in Spark side. WDYT?

Reply:
Yup, that was the initial proposal but I switched this to `true` long ago to make this PR sound safer and more compelling. I would like to change it to `true` if it's fine to you, @gatorsmile, @viirya and @cloud-fan too.

Reply:
I think it make a lot sense to default turn off parquet record level filtering.

Reply:
Let me change it to `false` for now. Thanks @jiangxb1987.

Reply:
BTW, for this default value, there was another small discussion here before - https://github.com/apache/spark/pull/14671#issuecomment-245972021 too.

Reply:
From the benchmark numbers, looks Spark-side filtering is always better. This default value should not change final results too. So a default value `false` should make sense.